utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/SwaggerTyped.scala

```diff
@@ -48,23 +48,29 @@ case class SwaggerObject(elementType: Map[PropertyName, SwaggerTyped], additiona
 
 case class SwaggerMap(valuesType: Option[SwaggerTyped]) extends SwaggerTyped
 
+//mapped to Unknown in type system
+case object SwaggerRecursiveSchemaFallback extends SwaggerTyped
+
 object SwaggerTyped {
 
   @tailrec
-  def apply(schema: Schema[_], swaggerRefSchemas: SwaggerRefSchemas): SwaggerTyped = schema match {
-    case objectSchema: ObjectSchema => SwaggerObject(objectSchema, swaggerRefSchemas)
-    case mapSchema: MapSchema => SwaggerObject(mapSchema, swaggerRefSchemas)
-    case arraySchema: ArraySchema => SwaggerArray(arraySchema, swaggerRefSchemas)
+  def apply(schema: Schema[_], swaggerRefSchemas: SwaggerRefSchemas, usedSchemas: Set[String] = Set.empty): SwaggerTyped = schema match {
```

Comment:
Can we have two methods - one public, without usedSchemas and second private? Do you see some case when we would like to pass it from the upper level?

Reply:
:+1: add second method